compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt

```diff
@@ -69,7 +106,7 @@ internal fun checkPropertyInitializer(
         }
     }
 
-    val isExpect = property.isExpect || modifierList?.modifiers?.any { it.token == KtTokens.EXPECT_KEYWORD } == true
+    val isExpect = property.isEffectivelyExpect(containingClass, modifierList, context)
```

Comment:
Since you made this change to check containing classes as well, I expected some more tests to be affected, i.e., for `EXPECTED_PROPERTY_INITIALIZER` or `EXPECTED_DELEGATED_PROPERTY`. But it looks like there's a hole in the tests. Code like this:
```kt
expect class A {
    class B {
        val c: Int = 1
    }
}
```
...has a `EXPECTED_PROPERTY_INITIALIZER` error on `A.B.c` property for FE1.0, but it currently does not for FIR. I assume that it would have an error now after your change. Can you add a test like this while you're at it? Also for `EXPECTED_DELEGATED_PROPERTY` (`val a: Int by lazy { 1 }`)

Reply:
The motivation is exactly what you described: https://github.com/JetBrains/kotlin/blob/master/compiler/testData/diagnostics/tests/multiplatform/headerClass/nestedClasses.kt#L11 (well, more nested actually)

Added lazy init as well.